src/Plugin/os2web/DataLookup/ServiceplatformenBase.php

```diff
@@ -321,9 +373,36 @@ protected function query(string $method, array $request): array {
         'status' => FALSE,
         'error' => $e->faultstring,
       ];
+    } finally {
+      // Remove temporary certificate file.
+      if (file_exists($localCertPath)) {
+        unlink($localCertPath);
+      }
     }
 
     return $response;
   }
 
+  /**
+   * Get certificate.
+   */
+  protected function getCertificate(): string {
+    $provider = $this->configuration['certificate_provider'] ?? NULL;
+    if (self::PROVIDER_TYPE_KEY === $provider) {
+      $keyId = $this->configuration['certificate_key'] ?? '';
+      $key = $this->keyRepository->getKey($keyId);
+      if (NULL === $key) {
+        throw new RuntimeException(sprintf('Cannot get key %s', $keyId));
+      }
+
+      return $key->getKeyValue();
+    }
+
+    $filename = 0 === $this->configuration['mode_selector']
+      ? $this->configuration['certfile']
+      : $this->configuration['certfile_test'];
```

Comment:
i suggest to simplify this line, took a while to understand what's happening here

Reply:
Clarifying comment added in https://github.com/OS2web/os2web_datalookup/pull/13/commits/92935b5a6409e6e776515926f71032bb60afc78d.